packages/jest-runner/src/run_test.js

```diff
@@ -106,22 +114,49 @@ export default (async function runTest(
     );
     const testCount =
       result.numPassingTests + result.numFailingTests + result.numPendingTests;
+
+    result.leaks = environmentLeakDetector;
     result.perfStats = {end: Date.now(), start};
     result.testFilePath = path;
     result.coverage = runtime.getAllCoverageInfo();
     result.sourceMaps = runtime.getSourceMapInfo();
     result.console = testConsole.getBuffer();
     result.skipped = testCount === result.numPendingTests;
     result.displayName = config.displayName;
+
     if (globalConfig.logHeapUsage) {
       if (global.gc) {
         global.gc();
       }
       result.memoryUsage = process.memoryUsage().heapUsed;
     }
+
     // Delay the resolution to allow log messages to be output.
-    return new Promise(resolve => setImmediate(() => resolve(result)));
+    await new Promise(resolve => setImmediate(resolve));
+
+    return result;
   } finally {
     await environment.teardown();
   }
-});
+}
+
+export default async function runTest(
+  path: Path,
+  globalConfig: GlobalConfig,
+  config: ProjectConfig,
+  resolver: Resolver,
+): Promise<TestResult> {
+  const result: TestResult = await runTestHelper(
+    path,
+    globalConfig,
+    config,
+    resolver,
+  );
+
+  // Resolve leak detector, so that the object is JSON serializable.
+  if (result.leaks instanceof LeakDetector) {
+    result.leaks = result.leaks.isLeaking();
```

Comment:
Ah, now I get it. This seems super hacky. How about instead of doing this we change `runTestHelper` to return `{leakDetector, result}` and turn `result.leaks` into a boolean?